webrtc/RTCRtpTransceiver-setCodecPreferences.html

```diff
@@ -0,0 +1,136 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCRtpTransceiver.prototype.setCodecPreferences</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  'use strict';
+
+  // Test is based on the following editor draft:
+  // https://w3c.github.io/webrtc-pc/archives/20170605/webrtc.html
+
+  /*
+    5.4.  RTCRtpTransceiver Interface
+      interface RTCRtpTransceiver {
+        ...
+        void setCodecPreferences(sequence<RTCRtpCodecCapability> codecs);
+      };
+
+      setCodecPreferences
+        - Setting codecs to an empty sequence resets codec preferences to any
+          default value.
+
+        - The codecs sequence passed into setCodecPreferences can only contain
+          codecs that are returned by RTCRtpSender.getCapabilities(kind) or
+          RTCRtpReceiver.getCapabilities(kind), where kind is the kind of the
+          RTCRtpTransceiver on which the method is called. Additionally, the
+          RTCRtpCodecParameters dictionary members cannot be modified. If
+          codecs does not fulfill these requirements, the user agent MUST throw
+          an InvalidAccessError.
+   */
+
+  test(() => {
+    const pc = new RTCPeerConnection();
+    const transceiver = pc.addTransceiver('audio');
+    const capabilities = RTCRtpSender.getCapabilities('audio');
+    transceiver.setCodecPreferences(capabilities.codecs);
+
+  }, `setCodecPreferences() on audio transceiver with codecs returned from RTCRtpSender.getCapabilities('audio') should succeed`);
```

Comment:
There is no simple way to prove it.  The method affects the SDP generated in `createOffer()` and `createAnswer()`. It is out of scope to test how browsers should reorder the codecs lines. Since the method is void we can only say it succeed if no error is thrown.